Allow JVM debug options to be configurable (#18758)
gradle/gradle: subprojects/docs/src/docs/userguide/reference/command_line_interface.adoc

`-Dorg.gradle.debug=true`::
Debug Gradle client (non-Daemon) process. Gradle will wait for you to attach a debugger at `localhost:5005` by default.

`-Dorg.gradle.debug.port=(port number)`::

> Specifies the port number on which process is listening when debug is enabled. _Default is `5005`._

Reviewer: ```suggestion
Specifies the port number to listen on when debug is enabled. _Default is `5005`._
```

`-Dorg.gradle.debug.server=true`::

> Then set to `true` and debugging enabled, Gradle will run the build with socked-attach mode of debugger. Otherwise socket-listen is used.

Reviewer: ```suggestion
If set to `true` and debugging is enabled, Gradle will run the build with the socket-attach mode of the debugger. Otherwise, the socket-listen mode is used.
```

`-Dorg.gradle.debug.suspend=true`::

> When set to `true` and debugging enabled, Gradle will suspend the virtual machine until a debugger is attached.

Reviewer: ```suggestion
When set to `true` and debugging is enabled, the JVM running Gradle will suspend until a debugger is attached.
```
